prime-dotnet-webapi/Services/CommunitySiteService.cs

```diff
@@ -263,6 +264,27 @@ private void UpdateVendors(CommunitySite current, CommunitySiteUpdateModel updat
             }
         }
 
+        private async Task UpdateIndividualDeviceProviders(int siteId, CommunitySiteUpdateModel updated)
+        {
+            var currentProviders = await _context.IndividualDeviceProviders
+                .Where(p => p.CommunitySiteId == siteId)
+                .ToListAsync();
+            _context.IndividualDeviceProviders.RemoveRange(currentProviders);
+
+
+            if (updated.IndividualDeviceProviders == null)
+            {
+                return;
+            }
```

Comment:
The current trend of our updates is if the update's collection is `null`, we don't touch the collection. We empty out the collection if we sent up an empty collection. Does that behavior fit with this feature? If so, this guard clause should move up to line 269.